CRM/Core/Payment/Elavon.php

```diff
@@ -142,71 +161,25 @@ public function doPayment(&$params, $component = 'contribute') {
     // Send to the payment processor using cURL
 
     $chHost = $host . '?xmldata=' . $xml;
-
-    $ch = curl_init($chHost);
-    if (!$ch) {
-      throw new PaymentProcessorException('Could not initiate connection to payment gateway', 9004);
-    }
-
-    curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, Civi::settings()->get('verifySSL') ? 2 : 0);
-    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, Civi::settings()->get('verifySSL'));
-    // return the result on success, FALSE on failure
-    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-    curl_setopt($ch, CURLOPT_TIMEOUT, 36000);
-    // set this for debugging -look for output in apache error log
-    //curl_setopt ($ch,CURLOPT_VERBOSE,1 );
-    // ensures any Location headers are followed
-    if (ini_get('open_basedir') == '') {
-      curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
-    }
-
-    // Send the data out over the wire
-    $responseData = curl_exec($ch);
-
-    // See if we had a curl error - if so tell 'em and bail out
-    // NOTE: curl_error does not return a logical value (see its documentation), but
-    // a string, which is empty when there was no error.
-    if ((curl_errno($ch) > 0) || (strlen(curl_error($ch)) > 0)) {
-      curl_close($ch);
-      $errorNum = curl_errno($ch);
-      $errorDesc = curl_error($ch);
-
-      // Paranoia - in the unlikley event that 'curl' errno fails
-      if ($errorNum == 0) {
-        $errorNum = 9005;
-      }
-
-      // Paranoia - in the unlikley event that 'curl' error fails
-      if (strlen($errorDesc) == 0) {
-        $errorDesc = 'Connection to payment gateway failed';
-      }
-      throw new PaymentProcessorException('Curl error - ' . $errorDesc . ' Try this link for more information http://curl.haxx.se/docs/sslcerts.html', $errorNum);
-    }
-
-    // If null data returned - tell 'em and bail out
-    // NOTE: You will not necessarily get a string back, if the request failed for
-    // any reason, the return value will be the boolean false.
-    if (($responseData === FALSE) || (strlen($responseData) == 0)) {
-      curl_close($ch);
-      throw new PaymentProcessorException('Error: Connection to payment gateway failed - no data returned.', 9006);
-    }
+    $responseData = $this->getGuzzleClient()->post($chHost, [
+      'curl' => [
+        CURLOPT_RETURNTRANSFER => TRUE,
+        CURLOPT_TIMEOUT => 36000,
+        CURLOPT_SSL_VERIFYHOST => Civi::settings()->get('verifySSL') ? 2 : 0,
+        CURLOPT_SSL_VERIFYPEER => Civi::settings()->get('verifySSL'),
+      ],
+    ])->getBody();
```

Comment:
@seamuslee001 is it required to set the `CURLOPT_FOLLOWLOCATION` if as per L160 above?

Reply:
ok have updated to it @monishdeb